compiler/rustc_ty_utils/src/assoc.rs

```diff
@@ -193,7 +219,50 @@ fn associated_item_for_impl_trait_in_trait(
     let span = tcx.def_span(opaque_ty_def_id);
     let trait_assoc_ty =
         tcx.at(span).create_def(trait_def_id.expect_local(), DefPathData::ImplTraitAssocTy);
-    trait_assoc_ty.def_id()
+
```

Comment:
Each one of these feedings should probably explain its correctness or why it's done the way it's done (or the FIXMEs need more words).

Reply:
Done